content/docs/guides/branch-restore.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Branch Restore
+subtitle: Learn how to revert changes or recover lost data using Neon Branch Restore, with Time Travel Assist to query historical data as part of your restore workflow.
+enableTableOfContents: true
+---
+
+With Neon's copy-on-write branch creation capability, just as you can instantly create branches from any point in your history retention window, you can just as easily restore a branch to an earlier state in that history. You can also use Time Travel Assist to help troubleshoot issues: run read-only queries against any point in your history retention window as a way to pinpoint the exact moment you need to restore to, before you proceed.
+
+## How it works
+
+### Restore from history
+
+The restore operation lets you revert the state of a selected branch to an earlier point in time. For example, revert to just before some corruption or loss of data.
+
+![branch restore to timestamp](/docs/guides/branch_restore_time.png)
+
+By default, your history retention is set to 7 days. You can revert a branch to any time within that configured [retention window](/docs/manage/projects#configure-history-retention), down to the millisecond.
+
+A few key points to keep in mind about the restore operation:
+- [Current data is overwritten](#overwrite-not-a-merge)
+- [Restore backups are created automatically](#automatic-backups)
+- [All databases on a branch are updated](#changes-apply-to-all-databases)
+- [Connections are temporarily interrupted](#connections-temporarily-interrupted)
+
+#### Overwrite, not a merge
+
+It is important to understand that whenever you restore a branch, you are performing a _complete_ overwrite, not a merge or refresh. Everything on your current branch, data and schema, is replaced with the contents from the historical source. All interim data from the selected restore point onwards is removed from the branch.
+
+#### Automatic backups
+
+In case you need to rollback the update, Neon preserves the branch's final state before the restore operation in an automatically created backup branch, which takes the following format:
```

Comment:
```suggestion
In case you need to rollback a restore, Neon preserves the branch's final state before the restore operation in an automatically created backup branch, which takes the following format:
```